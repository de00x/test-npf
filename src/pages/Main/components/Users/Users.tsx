import { FC } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { deleteUser } from '../../../../store/slices/usersSlice'
import { RootState } from '../../../../store/store'
import styles from './styles/Users.module.scss'

export const Users: FC = (): JSX.Element => {
  const users = useSelector((state: RootState) => state.users)
  const dispatch = useDispatch()

  return (
    <div className={styles.wrapper}>
      {users.map((user) => (
        <div
          className={styles.userContainer}
          onClick={() => dispatch(deleteUser(user.id))}
          key={user.id}
        >
          <div>
            name: <span>{user.name}</span>
          </div>
          <div>
            username: <span>{user.username}</span>
          </div>
          <div>
            email: <span>{user.email}</span>
          </div>
        </div>
      ))}
    </div>
  )
}
