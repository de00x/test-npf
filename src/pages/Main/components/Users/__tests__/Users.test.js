import { render } from '@testing-library/react'
import { useSelector } from 'react-redux'

import { Users } from '../Users'

jest.mock('react-redux')

const usersArray = [
  { id: 123, name: 'name', username: 'username', email: 'email' },
  { id: 321, name: 'namee', username: 'usernamee', email: 'emaile' },
]

describe('usersList', () => {
  it('should create usersList with empty users', () => {
    useSelector.mockReturnValue([])

    const component = render(<Users />)

    expect(component).toMatchSnapshot()
  })

  it('should create usersList with users', () => {
    useSelector.mockReturnValue(usersArray)

    const component = render(<Users />)

    expect(component).toMatchSnapshot()
  })

  it('should create Users', () => {
    useSelector.mockReturnValue(usersArray)
    const component = render(<Users />)

    expect(component).toMatchSnapshot()
  })
})
