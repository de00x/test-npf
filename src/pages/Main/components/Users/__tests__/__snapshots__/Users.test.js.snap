// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usersList should create Users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              name
            </span>
          </div>
          <div>
            username: 
            <span>
              username
            </span>
          </div>
          <div>
            email: 
            <span>
              email
            </span>
          </div>
        </div>
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              namee
            </span>
          </div>
          <div>
            username: 
            <span>
              usernamee
            </span>
          </div>
          <div>
            email: 
            <span>
              emaile
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            name
          </span>
        </div>
        <div>
          username: 
          <span>
            username
          </span>
        </div>
        <div>
          email: 
          <span>
            email
          </span>
        </div>
      </div>
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            namee
          </span>
        </div>
        <div>
          username: 
          <span>
            usernamee
          </span>
        </div>
        <div>
          email: 
          <span>
            emaile
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usersList should create usersList with empty users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usersList should create usersList with empty users 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              name
            </span>
          </div>
          <div>
            username: 
            <span>
              username
            </span>
          </div>
          <div>
            email: 
            <span>
              email
            </span>
          </div>
        </div>
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              namee
            </span>
          </div>
          <div>
            username: 
            <span>
              usernamee
            </span>
          </div>
          <div>
            email: 
            <span>
              emaile
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            name
          </span>
        </div>
        <div>
          username: 
          <span>
            username
          </span>
        </div>
        <div>
          email: 
          <span>
            email
          </span>
        </div>
      </div>
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            namee
          </span>
        </div>
        <div>
          username: 
          <span>
            usernamee
          </span>
        </div>
        <div>
          email: 
          <span>
            emaile
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usersList should create usersList with users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              name
            </span>
          </div>
          <div>
            username: 
            <span>
              username
            </span>
          </div>
          <div>
            email: 
            <span>
              email
            </span>
          </div>
        </div>
        <div
          class="userContainer"
        >
          <div>
            name: 
            <span>
              namee
            </span>
          </div>
          <div>
            username: 
            <span>
              usernamee
            </span>
          </div>
          <div>
            email: 
            <span>
              emaile
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            name
          </span>
        </div>
        <div>
          username: 
          <span>
            username
          </span>
        </div>
        <div>
          email: 
          <span>
            email
          </span>
        </div>
      </div>
      <div
        class="userContainer"
      >
        <div>
          name: 
          <span>
            namee
          </span>
        </div>
        <div>
          username: 
          <span>
            usernamee
          </span>
        </div>
        <div>
          email: 
          <span>
            emaile
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
